docs/book/src/plugins/creating-plugins.md

```diff
@@ -64,3 +64,46 @@ Feel free to check the implementation under:
 
 - deploy-image: <https://github.com/kubernetes-sigs/kubebuilder/tree/v3.7.0/pkg/plugins/golang/deploy-image/v1alpha1>
 - grafana: <https://github.com/kubernetes-sigs/kubebuilder/tree/v3.7.0/pkg/plugins/optional/grafana/v1alpha>
+
```

Comment:
WDYT about we add something like follow after the Overview section?

---
## When it is useful

- If you are looking for to create plugins which supports and work with another language
- If you would like to create helpers and integrations on top of the scaffolds done by the plugins provided by Kubebuilder.
- If you would like to have custmozed layouts according to your needs

## How the plugins can be used

Kubebuilder provides a set of plugins to scaffold the projects, to help you extend
and re-use its implementation as to provide additional features. For further information
see [Available Plugins](available-plugins.md).

Therefore, if you have a need you might want to propose a solution by adding a new plugin
which would be shipped with Kubebuilder by default. 

However, you might also want to have your own tool to address your specific scenarios and 
by taking advantage of what is provided by Kubebuilder as LIB alreadu for you are able to be
focus to address your needs and keep your solutions eaiser maintained. 

Note that you can use Kubebuilder as lib to import its plugins and then create your own
plugins that does customzations on top such as [Operator-SDK] does with the plugins [manifest](https://github.com/operator-framework/operator-sdk/tree/v1.23.0/internal/plugins/manifests/v2) and
[scorecard](https://github.com/operator-framework/operator-sdk/tree/v1.23.0/internal/plugins/scorecard/v2) for example to add its features. See [here](https://github.com/operator-framework/operator-sdk/blob/v1.23.0/internal/cmd/operator-sdk/cli/cli.go#L78-L160)

Another option implemented with the [Extensible CLI and Scaffolding Plugins - Phase 2][plugins-phase2-design-doc] is
to extend Kibebuilder as a LIB to cretae only a specific plugin that can be called and used with
Kubebuilder as well.

<aside class="note">
<H1> Plugins proposal docs</H1>

You can check the proposal documentation for better understand its motivations see the [Extensible CLI and Scaffolding Plugins: phase 1][plugins-phase1-design-doc],
the [Extensible CLI and Scaffolding Plugins: phase 1.5][plugins-phase1-design-doc-1.5] and the [Extensible CLI and Scaffolding Plugins - Phase 2][plugins-phase2-design-doc]
design docs. Also, you can check the [Plugins section][plugins-section].

</aside>